kf_lib_data_ingest/etl/load/load.py

```diff
@@ -456,19 +458,34 @@ def _run(self, target_entities):
 
             self.prev_entity = None
             total = len(entities)
+
+            if self.use_async:
+                ex = concurrent.futures.ThreadPoolExecutor()
+
             for i, entity in enumerate(entities):
                 entity_id = entity['id']
                 endpoint = entity['endpoint']
                 body = entity['properties']
                 links = entity['links']
 
+                verb = 'Queueing' if self.use_async else 'Loading'
                 self.logger.info(
-                    f'Loading {entity_type} {entity_id} (#{i+1} of {total})'
-                )
-                self._load_entity(
-                    entity_type, endpoint, entity_id, body, links
+                    f'{verb} {entity_type} {entity_id} (#{i+1} of {total})'
                 )
 
+                if self.use_async and not self.resume_from:
```

Comment:
Maybe you could do something like this so you don't have to check `use_async` twice and call shutdown:

```python
if self.use_async and not self.resume_from:
    with ThreadPoolExecutor() as ex:
        ex.submit(
            self._load_entity,
            entity_type, endpoint, entity_id, body, links
        )
```

Reply:
You need the creation of the pool to happen once at the start. If you create one with a context manager it automatically calls shutdown as soon as the context ends, so what you have there would not actually run anything in parallel.

I can remove lines 462 and 486 if you want, which would make it always create a pool and then always shut it down regardless of whether anything is going to be added to it. 

I can also put the whole block from line 462 until 487 inside of a context manager, but then that would be an additional nest indenting everything in order to save one line calling shutdown.

Reply:
Oh woops yea, my bad. I think its fine as is